### ISSUE TITLE
Add error messages to indicate why a factory is red - and what is triggering the error

### ISSUE THREAD

**Issue by samsmithnz** [Maelstromeous/satisfactory-factories#72]
I just spent about 15 minutes trying to figure out one of my factories was red. My scenario was that my steel factory was producing 15 stators, sending 10 to the iron factory, and 5 to the automated wiring factory. It turned out that my automated wiring factory was pulling 5 stators from the iron factory, not the steel factory - but this was confusing, as everything was green - except the iron factory, that was freaking out that there weren't enough stators - except this wasn't visibly obvious. It doesn't show that anything is wrong with the iron factory. 

To reproduce.
1. Create steel factory, export 15 stators
2. Create iron factory, import 15 stators from steel factory
3. Create Automated Wiring factory, import 5 stators from iron factory
4. Return to iron factory, click the trim button to reduce the 15 stators to 10. 
5. Review iron factory and see if you can see what is wrong.

**Comment by Maelstromeous** [comment]
Yeah that would be much better UX I completely agree

**Comment by Maelstromeous** [comment]
And now you can create a link to show me exactly the issue 😀

**Comment by samsmithnz** [comment]
The link to the broken factory! 

https://satisfactory-factories.app/share/melodic-abiding-agent

**Comment by Maelstromeous** [comment]
Not quite sure how you managed this but somehow the Iron Factory has an invisible dependency **from** the steel factory for 5 stators, but that is not showing up in the Exports tab because you don't have Statora as a product in the Iron Factory.

```JSON
"dependencies": {
    "requests": {
      "6842": [
        {
          "part": "Motor",
          "amount": 5
        }
      ],
      "9032": [
        {
          "part": "ModularFrameHeavy",
          "amount": 1
        },
        {
          "part": "Stator",
          "amount": 5
        }
      ]
    },
    "metrics": {
      "Motor": {
        "part": "Motor",
        "request": 5,
        "supply": 5,
        "isRequestSatisfied": true,
        "difference": 0
      },
      "ModularFrameHeavy": {
        "part": "ModularFrameHeavy",
        "request": 1,
        "supply": 1,
        "isRequestSatisfied": true,
        "difference": 0
      },
      "Stator": {
        "part": "Stator",
        "request": 5,
        "supply": 0,
        "isRequestSatisfied": false,
        "difference": -5
      }
    }
```

Strangely, this is not showing up in Exports.

**Comment by Maelstromeous** [comment]
Yeah Space Elevator Part 2 has 5 stators requesting from Iron. What must have happened is somehow you had them as a product at some point but the code didn't delete the invalid request 🤔

**Comment by Maelstromeous** [comment]
What's even weirder is that the entire dependency tree is rebuilt basically upon every single change as we have to check for validity. 

Could you try to step by step reproduce this so I can walk through how it got into this error state?

**Comment by Maelstromeous** [comment]
Also if you can provide a simplier example that would be much nicer. The ability for us to have tabs to load different shares is growing by the minute... I may have to make it happen just to keep on top of the debugging... already had to "backup" my own factory for my playthrough to a share link so I don't lose my place 😅

**Comment by Maelstromeous** [comment]
Related: #107

**Comment by samsmithnz** [comment]
The step by step is in the original post. It's mostly related to the trim in step 4, and not showing the stator deficiency.

**Comment by Maelstromeous** [comment]
So this is an issue of two parts, #107 is to fix the bug with the import not being removed, need to fully nail down a simple repro step for that, this issue will contain the message to the user why the factory is red, in addition to the red bordering I've implemented.

**Comment by Maelstromeous** [comment]
Super ceded by #229
